packages/graphql/src/translate/queryAST/factory/OperationFactory.ts

```diff
@@ -60,42 +62,33 @@ export class OperationsFactory {
         entityOrRel: ConcreteEntityAdapter | RelationshipAdapter,
         resolveTree: ResolveTree,
         context: Neo4jGraphQLTranslationContext
-    ): ReadOperation {
-        const entity = (
-            entityOrRel instanceof RelationshipAdapter ? entityOrRel.target : entityOrRel
-        ) as ConcreteEntityAdapter;
-        const projectionFields = { ...resolveTree.fieldsByTypeName[entity.name] };
+    ): ReadOperation | InterfaceReadOperation {
+        const entity = entityOrRel instanceof RelationshipAdapter ? entityOrRel.target : entityOrRel;
 
-        const whereArgs = (resolveTree.args.where || {}) as Record<string, unknown>;
-        const operation = new ReadOperation(entityOrRel, Boolean(resolveTree.args?.directed ?? true));
-        const fields = this.fieldFactory.createFields(entity, projectionFields, context);
-
-        const authFilters = this.authorizationFactory.createEntityAuthFilters(entity, ["READ"], context);
+        const relationship = entityOrRel instanceof RelationshipAdapter ? entityOrRel : undefined;
 
-        let filters: Filter[];
-        if (entityOrRel instanceof RelationshipAdapter) {
-            filters = this.filterFactory.createRelationshipFilters(entityOrRel, whereArgs);
+        if (!(entity instanceof ConcreteEntityAdapter)) {
+            return this.createInterfaceReadOperationAST({
```

Comment:
This creates a little confusion with the previous variable entityOrRel reassignments, as depending on the context, this condition 
`!(entity instanceof ConcreteEntityAdapter)` could be valid for `RelationshipAdapter | UnionAdapter | InterfaceAdapter` so it adds a lot of cognitive complexity, I think it's worth refactoring this piece now that we have clear what the real meaning is!

Reply:
mmm, I've changed a bit the conditions here so it is a bit easier to follow.

However, the bottom line is that if a relationship is provided, entity will be the target. Meaning that entity can only be of type ConcreateEntity, Union or Interface

I've moved the conditionals around to avoid the negative